pkg/apis/discovery/validation/validation.go

```diff
@@ -110,11 +110,11 @@ func validateEndpoints(endpoints []discovery.Endpoint, addrType discovery.Addres
 			}
 		}
 
-		topologyPath := idxPath.Child("topology")
-		if len(endpoint.Topology) > maxTopologyLabels {
-			allErrs = append(allErrs, field.TooMany(topologyPath, len(endpoint.Topology), maxTopologyLabels))
+		topologyPath := idxPath.Child("deprecatedTopology")
+		if len(endpoint.DeprecatedTopology) > maxTopologyLabels {
+			allErrs = append(allErrs, field.TooMany(topologyPath, len(endpoint.DeprecatedTopology), maxTopologyLabels))
 		}
-		allErrs = append(allErrs, metavalidation.ValidateLabels(endpoint.Topology, topologyPath)...)
+		allErrs = append(allErrs, metavalidation.ValidateLabels(endpoint.DeprecatedTopology, topologyPath)...)
```

Comment:
In my sketch I had

```
+               if _, found := endpoint.DeprecatedTopology[v1.LabelTopologyZone]; found {
+                       allErrs = append(allErrs, field.InternalError(topologyPath.Key(v1.LabelTopologyZone), fmt.Errorf("reserved key was not removed in conversion")))
+               }
```

Do you think that's a bad idea?

Reply:
This is the only comment to resolve still.

Did you figure out why the v1b1 controller, setting region, wasn't coming thru to v1 deprecatedTopology?  Highly likely it was PEBKAC, but would be good to verify

Reply:
I had a question about it initially and just forgot to add it back in once I resolved it.

With this branch, when v1b1 controller set region it came through in v1 deprecatedTopology. I didn't investigate too much further. What is PEBKAC?

Reply:
PEBKAC - Problem Exists Between Keyboard And Chair